### PR TITLE
verify the spdx licenses passed in to the license allowlist

### PR DESCRIPTION
**Pull request by josieang**

### CODE REVIEW

File: pkg/spdx/gen.go
Comment (josieang) [inline pkg/spdx/gen.go:1]: code for reading spdx expressions from https://github.com/spdx/license-list-data/blob/main/json/licenses.json into a file using go generate

**Review by another-rex — APPROVED** [review]
LGTM! Thanks!

**Comment by another-rex** [comment]
Can you add a test for the unrecognized code path?

**Comment by josieang** [comment]
> Can you add a test for the unrecognized code path?

done!

**Pull request merged**